signer/tests/integration/stacks_events_observer.rs

```diff
@@ -415,6 +421,20 @@ async fn test_new_blocks_sends_update_deposits_to_emily() {
 
     let bitcoin_txid = deposit_completed_event.outpoint.txid.to_string();
 
+    // Insert a dummy sweep transaction into the database. This will be retrieved by
+    // handle_completed_deposit, using the txid from the event to retrieve the tx fee.
+    let mut sweep = fake::Faker.fake_with_rng::<SweepTransaction, _>(&mut OsRng);
+    sweep.txid = deposit_completed_event.sweep_txid.into();
+    sweep.created_at_block_hash = deposit_completed_event.sweep_block_hash.into();
+    // A real sweep transaction would have the relevant deposits. For this test, we don't need them.
+    sweep.swept_deposits = vec![];
+    sweep.swept_withdrawals = vec![];
+    context
+        .get_storage_mut()
+        .write_sweep_transaction(&sweep)
+        .await
+        .expect("failed to insert dummy sweep transaction");
```

Comment:
Can't this all be removed?

Reply:
yeah, I had to swap it with a deposit request 🤦 . Done in https://github.com/stacks-network/sbtc/pull/1005/commits/2b6e492f8f4a0868f541cf59b8f33bec06c25e67